src/backend/gporca/libgpos/src/error/CMessageRepository.cpp

```diff
@@ -103,12 +103,12 @@ CMessageRepository::LookupMessage(CException exc, ELocale locale)
 //		Initialize global instance of message repository
 //
 //---------------------------------------------------------------------------
-GPOS_RESULT
+void
 CMessageRepository::Init()
 {
 	GPOS_ASSERT(NULL == m_repository);
 
-	CAutoMemoryPool amp;
+	CAutoMemoryPool amp(CAutoMemoryPool::ElcStrict);
```

Comment:
This change fixes the second case and does not depend on other changes. So I suggest to place it in a separate commit.

Reply:
As you can see, I added this argument to the constructor not only in this place, but also in many other places when initializing the ORCA, because a scenario similar to the second case can also occur in them.

Reply:
The constructor argument has been changed to avoid an exception in the destructor. It will be better to exclude raising exception in the destructor at all.

Reply:
done